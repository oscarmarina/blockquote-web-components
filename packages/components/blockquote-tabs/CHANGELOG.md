# Change Log

All notable changes to this project will be documented in this file.
See [Conventional Commits](https://conventionalcommits.org) for commit guidelines.

## [1.6.6](https://github.com/oscarmarina/blockquote-web-components/compare/@blockquote-web-components/blockquote-tabs@1.6.5...@blockquote-web-components/blockquote-tabs@1.6.6) (2025-08-13)

**Note:** Version bump only for package @blockquote-web-components/blockquote-tabs





## [1.6.5](https://github.com/oscarmarina/blockquote-web-components/compare/@blockquote-web-components/blockquote-tabs@1.6.4...@blockquote-web-components/blockquote-tabs@1.6.5) (2025-08-13)


### Bug Fixes

* **blockquote-tabs:** css logical properties ([db60988](https://github.com/oscarmarina/blockquote-web-components/commit/db60988becceb072eca29c74d6330227263d3d00))





## [1.6.4](https://github.com/oscarmarina/blockquote-web-components/compare/@blockquote-web-components/blockquote-tabs@1.6.3...@blockquote-web-components/blockquote-tabs@1.6.4) (2025-08-12)

**Note:** Version bump only for package @blockquote-web-components/blockquote-tabs





## [1.6.3](https://github.com/oscarmarina/blockquote-web-components/compare/@blockquote-web-components/blockquote-tabs@1.6.2...@blockquote-web-components/blockquote-tabs@1.6.3) (2025-07-20)

**Note:** Version bump only for package @blockquote-web-components/blockquote-tabs





## [1.6.2](https://github.com/oscarmarina/blockquote-web-components/compare/@blockquote-web-components/blockquote-tabs@1.6.1...@blockquote-web-components/blockquote-tabs@1.6.2) (2025-07-11)


### Bug Fixes

* **blockquote-tabs:** update all non-major dependencies ([497c66b](https://github.com/oscarmarina/blockquote-web-components/commit/497c66b9d07729e376646ca7e7223f49ca6e87d3))





## [1.6.1](https://github.com/oscarmarina/blockquote-web-components/compare/@blockquote-web-components/blockquote-tabs@1.6.0...@blockquote-web-components/blockquote-tabs@1.6.1) (2025-06-19)

**Note:** Version bump only for package @blockquote-web-components/blockquote-tabs





# [1.6.0](https://github.com/oscarmarina/blockquote-web-components/compare/@blockquote-web-components/blockquote-tabs@1.5.11...@blockquote-web-components/blockquote-tabs@1.6.0) (2025-04-16)


### Features

* **blockquote-tabs:** add useDefault ([63f1086](https://github.com/oscarmarina/blockquote-web-components/commit/63f1086034a1bd049eefd34d85751ec4d8ba9801))





## [1.5.11](https://github.com/oscarmarina/blockquote-web-components/compare/@blockquote-web-components/blockquote-tabs@1.5.10...@blockquote-web-components/blockquote-tabs@1.5.11) (2025-04-15)

**Note:** Version bump only for package @blockquote-web-components/blockquote-tabs





## [1.5.10](https://github.com/oscarmarina/blockquote-web-components/compare/@blockquote-web-components/blockquote-tabs@1.5.9...@blockquote-web-components/blockquote-tabs@1.5.10) (2025-03-25)


### Bug Fixes

* **blockquote-tabs:** add test ([547cac4](https://github.com/oscarmarina/blockquote-web-components/commit/547cac47600a4a10d629678ea43f4f6199283d57))





## [1.5.9](https://github.com/oscarmarina/blockquote-web-components/compare/@blockquote-web-components/blockquote-tabs@1.5.8...@blockquote-web-components/blockquote-tabs@1.5.9) (2025-03-14)

**Note:** Version bump only for package @blockquote-web-components/blockquote-tabs





## [1.5.8](https://github.com/oscarmarina/blockquote-web-components/compare/@blockquote-web-components/blockquote-tabs@1.5.7...@blockquote-web-components/blockquote-tabs@1.5.8) (2025-02-27)

**Note:** Version bump only for package @blockquote-web-components/blockquote-tabs





## [1.5.7](https://github.com/oscarmarina/blockquote-web-components/compare/@blockquote-web-components/blockquote-tabs@1.5.6...@blockquote-web-components/blockquote-tabs@1.5.7) (2025-02-26)

**Note:** Version bump only for package @blockquote-web-components/blockquote-tabs





## [1.5.6](https://github.com/oscarmarina/blockquote-web-components/compare/@blockquote-web-components/blockquote-tabs@1.5.5...@blockquote-web-components/blockquote-tabs@1.5.6) (2024-12-30)

**Note:** Version bump only for package @blockquote-web-components/blockquote-tabs





## [1.5.5](https://github.com/oscarmarina/blockquote-web-components/compare/@blockquote-web-components/blockquote-tabs@1.5.4...@blockquote-web-components/blockquote-tabs@1.5.5) (2024-12-26)


### Bug Fixes

* **blockquote-tabs:** update dependencies ([eccb7ba](https://github.com/oscarmarina/blockquote-web-components/commit/eccb7ba2a41d94b6b9a58d08ccc4950742c0860b))





## [1.5.4](https://github.com/oscarmarina/blockquote-web-components/compare/@blockquote-web-components/blockquote-tabs@1.5.3...@blockquote-web-components/blockquote-tabs@1.5.4) (2024-12-06)

**Note:** Version bump only for package @blockquote-web-components/blockquote-tabs





## [1.5.3](https://github.com/oscarmarina/blockquote-web-components/compare/@blockquote-web-components/blockquote-tabs@1.5.2...@blockquote-web-components/blockquote-tabs@1.5.3) (2024-11-28)

**Note:** Version bump only for package @blockquote-web-components/blockquote-tabs





## [1.5.2](https://github.com/oscarmarina/blockquote-web-components/compare/@blockquote-web-components/blockquote-tabs@1.5.1...@blockquote-web-components/blockquote-tabs@1.5.2) (2024-11-24)

**Note:** Version bump only for package @blockquote-web-components/blockquote-tabs





## [1.5.1](https://github.com/oscarmarina/blockquote-web-components/compare/@blockquote-web-components/blockquote-tabs@1.5.0...@blockquote-web-components/blockquote-tabs@1.5.1) (2024-11-11)

**Note:** Version bump only for package @blockquote-web-components/blockquote-tabs





# [1.5.0](https://github.com/oscarmarina/blockquote-web-components/compare/@blockquote-web-components/blockquote-tabs@1.4.11...@blockquote-web-components/blockquote-tabs@1.5.0) (2024-11-08)


### Features

* **blockquote-tabs:** update exports ([15254db](https://github.com/oscarmarina/blockquote-web-components/commit/15254db93682ddb12d7d83044a304a587fa32eb1))





## [1.4.11](https://github.com/oscarmarina/blockquote-web-components/compare/@blockquote-web-components/blockquote-tabs@1.4.10...@blockquote-web-components/blockquote-tabs@1.4.11) (2024-11-08)

**Note:** Version bump only for package @blockquote-web-components/blockquote-tabs





## [1.4.10](https://github.com/oscarmarina/blockquote-web-components/compare/@blockquote-web-components/blockquote-tabs@1.4.9...@blockquote-web-components/blockquote-tabs@1.4.10) (2024-10-18)

**Note:** Version bump only for package @blockquote-web-components/blockquote-tabs





## [1.4.9](https://github.com/oscarmarina/blockquote-web-components/compare/@blockquote-web-components/blockquote-tabs@1.4.8...@blockquote-web-components/blockquote-tabs@1.4.9) (2024-10-17)

**Note:** Version bump only for package @blockquote-web-components/blockquote-tabs





## [1.4.8](https://github.com/oscarmarina/blockquote-web-components/compare/@blockquote-web-components/blockquote-tabs@1.4.7...@blockquote-web-components/blockquote-tabs@1.4.8) (2024-10-05)

**Note:** Version bump only for package @blockquote-web-components/blockquote-tabs





## [1.4.7](https://github.com/oscarmarina/blockquote-web-components/compare/@blockquote-web-components/blockquote-tabs@1.4.6...@blockquote-web-components/blockquote-tabs@1.4.7) (2024-09-13)

**Note:** Version bump only for package @blockquote-web-components/blockquote-tabs





## [1.4.6](https://github.com/oscarmarina/blockquote-web-components/compare/@blockquote-web-components/blockquote-tabs@1.4.5...@blockquote-web-components/blockquote-tabs@1.4.6) (2024-09-13)

**Note:** Version bump only for package @blockquote-web-components/blockquote-tabs





## [1.4.5](https://github.com/oscarmarina/blockquote-web-components/compare/@blockquote-web-components/blockquote-tabs@1.4.4...@blockquote-web-components/blockquote-tabs@1.4.5) (2024-09-06)

**Note:** Version bump only for package @blockquote-web-components/blockquote-tabs





## [1.4.4](https://github.com/oscarmarina/blockquote-web-components/compare/@blockquote-web-components/blockquote-tabs@1.4.3...@blockquote-web-components/blockquote-tabs@1.4.4) (2024-07-26)


### Bug Fixes

* **blockquote-tabs:** update event-listeners ([1986199](https://github.com/oscarmarina/blockquote-web-components/commit/1986199d8286cb09aed453e07bddb9799b46e1e4))





## [1.4.3](https://github.com/oscarmarina/blockquote-web-components/compare/@blockquote-web-components/blockquote-tabs@1.4.2...@blockquote-web-components/blockquote-tabs@1.4.3) (2024-07-26)

**Note:** Version bump only for package @blockquote-web-components/blockquote-tabs





## [1.4.2](https://github.com/oscarmarina/blockquote-web-components/compare/@blockquote-web-components/blockquote-tabs@1.4.0...@blockquote-web-components/blockquote-tabs@1.4.2) (2024-07-25)

**Note:** Version bump only for package @blockquote-web-components/blockquote-tabs





## [1.4.1](https://github.com/oscarmarina/blockquote-web-components/compare/@blockquote-web-components/blockquote-tabs@1.4.0...@blockquote-web-components/blockquote-tabs@1.4.1) (2024-07-25)

**Note:** Version bump only for package @blockquote-web-components/blockquote-tabs





# [1.4.0](https://github.com/oscarmarina/blockquote-web-components/compare/@blockquote-web-components/blockquote-tabs@1.3.3...@blockquote-web-components/blockquote-tabs@1.4.0) (2024-06-29)


### Features

* **blockquote-tabs:** instance field with an arrow function ([facd5d9](https://github.com/oscarmarina/blockquote-web-components/commit/facd5d9d53101b70f7bd68bafde2890bfca18be5))





## [1.3.3](https://github.com/oscarmarina/blockquote-web-components/compare/@blockquote-web-components/blockquote-tabs@1.3.2...@blockquote-web-components/blockquote-tabs@1.3.3) (2024-06-23)

**Note:** Version bump only for package @blockquote-web-components/blockquote-tabs





## [1.3.2](https://github.com/oscarmarina/blockquote-web-components/compare/@blockquote-web-components/blockquote-tabs@1.3.1...@blockquote-web-components/blockquote-tabs@1.3.2) (2024-06-21)

**Note:** Version bump only for package @blockquote-web-components/blockquote-tabs





## [1.3.1](https://github.com/oscarmarina/blockquote-web-components/compare/@blockquote-web-components/blockquote-tabs@1.3.0...@blockquote-web-components/blockquote-tabs@1.3.1) (2024-06-06)

**Note:** Version bump only for package @blockquote-web-components/blockquote-tabs

# [1.3.0](https://github.com/oscarmarina/blockquote-web-components/compare/@blockquote-web-components/blockquote-tabs@1.2.0...@blockquote-web-components/blockquote-tabs@1.3.0) (2024-03-09)

### Features

- **blockquote-tabs:** move index to src folder ([1551306](https://github.com/oscarmarina/blockquote-web-components/commit/1551306c5f920e347f3681d8db1349c08a0d5fe0))

# [1.2.0](https://github.com/oscarmarina/blockquote-web-components/compare/@blockquote-web-components/blockquote-tabs@1.1.8...@blockquote-web-components/blockquote-tabs@1.2.0) (2024-03-08)

### Features

- **blockquote-tabs:** use absolute redirect ([962138e](https://github.com/oscarmarina/blockquote-web-components/commit/962138e88f9e5f22677bab7302db2b4e37f90881))

## [1.1.8](https://github.com/oscarmarina/blockquote-web-components/compare/@blockquote-web-components/blockquote-tabs@1.1.7...@blockquote-web-components/blockquote-tabs@1.1.8) (2024-01-29)

**Note:** Version bump only for package @blockquote-web-components/blockquote-tabs

## [1.1.7](https://github.com/oscarmarina/blockquote-web-components/compare/@blockquote-web-components/blockquote-tabs@1.1.6...@blockquote-web-components/blockquote-tabs@1.1.7) (2024-01-28)

**Note:** Version bump only for package @blockquote-web-components/blockquote-tabs

## [1.1.6](https://github.com/oscarmarina/blockquote-web-components/compare/@blockquote-web-components/blockquote-tabs@1.1.5...@blockquote-web-components/blockquote-tabs@1.1.6) (2024-01-26)

**Note:** Version bump only for package @blockquote-web-components/blockquote-tabs

## [1.1.5](https://github.com/oscarmarina/blockquote-web-components/compare/@blockquote-web-components/blockquote-tabs@1.1.4...@blockquote-web-components/blockquote-tabs@1.1.5) (2024-01-17)

**Note:** Version bump only for package @blockquote-web-components/blockquote-tabs

## [1.1.4](https://github.com/oscarmarina/blockquote-web-components/compare/@blockquote-web-components/blockquote-tabs@1.1.3...@blockquote-web-components/blockquote-tabs@1.1.4) (2024-01-16)

**Note:** Version bump only for package @blockquote-web-components/blockquote-tabs

## [1.1.3](https://github.com/oscarmarina/blockquote-web-components/compare/@blockquote-web-components/blockquote-tabs@1.1.2...@blockquote-web-components/blockquote-tabs@1.1.3) (2024-01-01)

### Bug Fixes

- **blockquote-tabs:** update dependencies ([df61fb7](https://github.com/oscarmarina/blockquote-web-components/commit/df61fb74f5e3e689ed2a23be4ac766aa323074c0))

## [1.1.2](https://github.com/oscarmarina/blockquote-web-components/compare/@blockquote-web-components/blockquote-tabs@1.1.1...@blockquote-web-components/blockquote-tabs@1.1.2) (2023-11-26)

**Note:** Version bump only for package @blockquote-web-components/blockquote-tabs

## [1.1.1](https://github.com/oscarmarina/blockquote-web-components/compare/@blockquote-web-components/blockquote-tabs@1.1.0...@blockquote-web-components/blockquote-tabs@1.1.1) (2023-11-25)

**Note:** Version bump only for package @blockquote-web-components/blockquote-tabs

# [1.1.0](https://github.com/oscarmarina/blockquote-web-components/compare/@blockquote-web-components/blockquote-tabs@1.0.5...@blockquote-web-components/blockquote-tabs@1.1.0) (2023-11-18)

### Features

- **blockquote-tabs:** lit3 ([5d3a3dc](https://github.com/oscarmarina/blockquote-web-components/commit/5d3a3dc65f9c7b1a60c70176135e8e1e05c19fd9))

## [1.0.5](https://github.com/oscarmarina/blockquote-web-components/compare/@blockquote-web-components/blockquote-tabs@1.0.4...@blockquote-web-components/blockquote-tabs@1.0.5) (2023-10-12)

**Note:** Version bump only for package @blockquote-web-components/blockquote-tabs

## [1.0.4](https://github.com/oscarmarina/blockquote-web-components/compare/@blockquote-web-components/blockquote-tabs@1.0.3...@blockquote-web-components/blockquote-tabs@1.0.4) (2023-09-06)

**Note:** Version bump only for package @blockquote-web-components/blockquote-tabs

## [1.0.3](https://github.com/oscarmarina/blockquote-web-components/compare/@blockquote-web-components/blockquote-tabs@1.0.2...@blockquote-web-components/blockquote-tabs@1.0.3) (2023-07-14)

**Note:** Version bump only for package @blockquote-web-components/blockquote-tabs

## [1.0.2](https://github.com/oscarmarina/blockquote-web-components/compare/@blockquote-web-components/blockquote-tabs@1.0.1...@blockquote-web-components/blockquote-tabs@1.0.2) (2023-07-06)

### Bug Fixes

- **blockquote-tabs:** update custom-elements ([f75de4b](https://github.com/oscarmarina/blockquote-web-components/commit/f75de4b6674c9b1fdd77a28a54790d59dec5689f))

## [1.0.1](https://github.com/oscarmarina/blockquote-web-components/compare/@blockquote-web-components/blockquote-tabs@1.0.0...@blockquote-web-components/blockquote-tabs@1.0.1) (2023-07-03)

### Bug Fixes

- **blockquote-tabs:** exports and lighthouse ([f4a9bd1](https://github.com/oscarmarina/blockquote-web-components/commit/f4a9bd10858b7caa9b5240c64defc5894fb2e118))

# 1.0.0 (2023-07-02)

### Features

- **blockquote-tabs:** new element ([c36a22b](https://github.com/oscarmarina/blockquote-web-components/commit/c36a22b2c54e4af3459b410b564d88caf64ef5f4))
