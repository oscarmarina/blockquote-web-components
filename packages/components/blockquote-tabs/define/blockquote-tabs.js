import { BlockquoteTabs } from '../src/BlockquoteTabs.js';
import './blockquote-tab.js';
import './blockquote-tabpanel.js';

window.customElements.define(BlockquoteTabs.is, BlockquoteTabs);
