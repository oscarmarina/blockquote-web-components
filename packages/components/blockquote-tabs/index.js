export { BlockquoteTabs } from './src/BlockquoteTabs.js';
export { BlockquoteTab } from './src/tab/BlockquoteTab.js';
export { BlockquoteTabPanel } from './src/tabpanel/BlockquoteTabPanel.js';
