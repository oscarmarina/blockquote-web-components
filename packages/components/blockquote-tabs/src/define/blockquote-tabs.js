import {BlockquoteTabs} from '../BlockquoteTabs.js';
import './blockquote-tab.js';
import './blockquote-tabpanel.js';

window.customElements.define('blockquote-tabs', BlockquoteTabs);
