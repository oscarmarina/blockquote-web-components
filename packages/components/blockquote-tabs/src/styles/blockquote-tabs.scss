:host {
  --_unselected-color: var(--blockquote-tabs-unselected-color, rgb(234 234 234));
  --_scroll-gradient-start-color: var(--blockquote-tabs-scroll-gradient-start-color, rgba(255, 255, 255, 0));
  --_scroll-gradient-end-color: var(--blockquote-tabs-scroll-gradient-end-color, rgba(252, 252, 252, 1));
  --_scroll-arrow-color: var(--blockquote-tabs-scroll-arrow-color, rgb(94, 94, 94));

  contain: content;
  display: block;
  box-sizing: border-box;
  color: #202020;
}

:host([hidden]),
[hidden] {
  display: none !important;
}

*,
*::before,
*::after {
  box-sizing: inherit;
}

.hold {
  position: relative;
  display: flex;

  .separator {
    position: absolute;
    z-index: -1;
    bottom: 0;
    height: 1px;
    left: 0;
    right: 0;
    background-color: var(--_unselected-color);
  }

  .indicator {
    position: absolute;
    width: 1rem;
    pointer-events: none;
    top: 0;
    bottom: 1px;
    left: 0;
    box-shadow: rgb(255 255 255) 4px 0 12px 4px inset, rgba(210, 210, 210, 1) 4px 0 4px -4px;
    background-color: rgba(234, 234, 234, 0.5);
    transition: opacity 92ms ease-in-out;
    opacity: 0;
    background-clip: padding-box;

    &.show-indicator {
      opacity: 1;
    }

    & + .indicator {
      left: auto;
      right: 0;
      transform: scale(-1);
    }
  }
}

.scroll-content {
  width: 100%;
  overflow: auto hidden;
  -webkit-overflow-scrolling: touch;
  scrollbar-width: none;
  overscroll-behavior: none;

  &::-webkit-scrollbar {
    display: none;
  }

  &:focus-visible {
    outline: #9e9e9e dashed 0.0625rem;
    outline-offset: -2px;
  }

  &:focus:not(:focus-visible) {
    outline: none;
  }
}

[role="tablist"] {
  display: flex;
}
