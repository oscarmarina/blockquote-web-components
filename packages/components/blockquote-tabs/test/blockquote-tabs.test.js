/* eslint-disable lit-a11y/no-autofocus */
/* eslint-disable import/no-extraneous-dependencies */
import { html, fixture, assert, fixtureCleanup, aTimeout } from '@open-wc/testing';

import '../define/blockquote-tabs.js';

suite('BlockquoteTabs', () => {
  let el;

  teardown(() => fixtureCleanup());

  suite('Default', () => {
    setup(async () => {
      el = await fixture(html`
        <blockquote-tabs>
          <blockquote-tab id="tab-1">Tab 1</blockquote-tab>
          <blockquote-tab id="tab-2">Tab 2</blockquote-tab>
          <blockquote-tab id="tab-3">Tab 3</blockquote-tab>
          <blockquote-tabpanel aria-labelledby="tab-1"><p>Panel 1</p></blockquote-tabpanel>
          <blockquote-tabpanel aria-labelledby="tab-2"><p>Panel 2</p></blockquote-tabpanel>
          <blockquote-tabpanel aria-labelledby="tab-3"><p>Panel 3</p></blockquote-tabpanel>
        </blockquote-tabs>
      `);
      await el.updateComplete;
    });

    suite('Semantic Dom and a11y', () => {
      test('SHADOW DOM - Structure test', async () => {
        await assert.shadowDom.equalSnapshot(el, { ignoreAttributes: ['id'] });
        await aTimeout(34);
      });

      test('LIGHT DOM - Structure test', async () => {
        await assert.lightDom.equalSnapshot(el, { ignoreAttributes: ['id'] });
        await aTimeout(34);
      });

      test('a11y', async () => {
        await assert.isAccessible(el);
        await aTimeout(34);
      });

      test('Click on tab updates selected tab', async () => {
        assert.equal(el.selected, 1);
        const tab = el.querySelectorAll('[role="tab"]')[2];
        tab.dispatchEvent(new MouseEvent('click', { bubbles: true, cancelable: true }));
        await el.updateComplete;
        assert.equal(el.selected, 3);
        await aTimeout(34);
      });

      test('ArrowRight on tab updates selected tab', async () => {
        assert.equal(el.selected, 1);
        const tab = el.querySelector('[role="tab"]');
        tab.dispatchEvent(
          new KeyboardEvent('keydown', { key: 'ArrowRight', bubbles: true, cancelable: true }),
        );
        await el.updateComplete;
        assert.equal(el.selected, 2);
        await aTimeout(34);
      });

      test('ArrowLeft on tab updates selected tab', async () => {
        assert.equal(el.selected, 1);
        const tab = el.querySelector('[role="tab"]');
        tab.dispatchEvent(
          new KeyboardEvent('keydown', { key: 'ArrowLeft', bubbles: true, cancelable: true }),
        );
        await el.updateComplete;
        assert.equal(el.selected, 3);
        await aTimeout(34);
      });
    });
  });

  suite('Autofocus', () => {
    setup(async () => {
      el = await fixture(html`
        <blockquote-tabs autofocus>
          <blockquote-tab id="tab-1">Tab 1</blockquote-tab>
          <blockquote-tab id="tab-2">Tab 2</blockquote-tab>
          <blockquote-tab id="tab-3">Tab 3</blockquote-tab>
          <blockquote-tabpanel aria-labelledby="tab-1"><p>Panel 1</p></blockquote-tabpanel>
          <blockquote-tabpanel aria-labelledby="tab-2"><p>Panel 2</p></blockquote-tabpanel>
          <blockquote-tabpanel aria-labelledby="tab-3"><p>Panel 3</p></blockquote-tabpanel>
        </blockquote-tabs>
      `);
      await el.updateComplete;
    });

    suite('Semantic Dom and a11y', () => {
      test('SHADOW DOM - Structure test', async () => {
        await assert.shadowDom.equalSnapshot(el, { ignoreAttributes: ['id'] });
        await aTimeout(34);
      });

      test('LIGHT DOM - Structure test', async () => {
        await assert.lightDom.equalSnapshot(el, { ignoreAttributes: ['id'] });
        await aTimeout(34);
      });

      test('a11y', async () => {
        await assert.isAccessible(el);
        await aTimeout(34);
      });
    });
  });

  suite('Selected', () => {
    setup(async () => {
      el = await fixture(html`
        <blockquote-tabs selected="2">
          <blockquote-tab id="tab-1">Tab 1</blockquote-tab>
          <blockquote-tab id="tab-2">Tab 2</blockquote-tab>
          <blockquote-tab id="tab-3">Tab 3</blockquote-tab>
          <blockquote-tabpanel aria-labelledby="tab-1"><p>Panel 1</p></blockquote-tabpanel>
          <blockquote-tabpanel aria-labelledby="tab-2"><p>Panel 2</p></blockquote-tabpanel>
          <blockquote-tabpanel aria-labelledby="tab-3"><p>Panel 3</p></blockquote-tabpanel>
        </blockquote-tabs>
      `);
      await el.updateComplete;
    });

    suite('Semantic Dom and a11y', () => {
      test('SHADOW DOM - Structure test', async () => {
        await assert.shadowDom.equalSnapshot(el, { ignoreAttributes: ['id'] });
        await aTimeout(34);
      });

      test('LIGHT DOM - Structure test', async () => {
        await assert.lightDom.equalSnapshot(el, { ignoreAttributes: ['id'] });
        await aTimeout(34);
      });

      test('a11y', async () => {
        await assert.isAccessible(el);
        await aTimeout(34);
      });
    });
  });
});
