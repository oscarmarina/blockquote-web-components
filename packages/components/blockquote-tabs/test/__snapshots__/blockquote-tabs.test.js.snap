// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`BlockquoteTabs > Autofocus > Semantic Dom and a11y > LIGHT DOM - Structure test > LIGHT DOM 1`] = `
"<blockquote-tabs autofocus="">
  <blockquote-tab
    aria-selected="false"
    role="tab"
    slot="tab"
    tabindex="-1"
  >
    Tab 1
  </blockquote-tab>
  <blockquote-tab
    aria-selected="false"
    role="tab"
    slot="tab"
    tabindex="-1"
  >
    Tab 2
  </blockquote-tab>
  <blockquote-tab
    aria-selected="false"
    role="tab"
    slot="tab"
    tabindex="-1"
  >
    Tab 3
  </blockquote-tab>
  <blockquote-tabpanel
    aria-hidden="true"
    aria-labelledby="tab-1"
    role="tabpanel"
    slot="tabpanel"
    tabindex="0"
  >
    <p>
      Panel 1
    </p>
  </blockquote-tabpanel>
  <blockquote-tabpanel
    aria-hidden="true"
    aria-labelledby="tab-2"
    role="tabpanel"
    slot="tabpanel"
    tabindex="0"
  >
    <p>
      Panel 2
    </p>
  </blockquote-tabpanel>
  <blockquote-tabpanel
    aria-hidden="true"
    aria-labelledby="tab-3"
    role="tabpanel"
    slot="tabpanel"
    tabindex="0"
  >
    <p>
      Panel 3
    </p>
  </blockquote-tabpanel>
</blockquote-tabs>
"
`;

exports[`BlockquoteTabs > Autofocus > Semantic Dom and a11y > SHADOW DOM - Structure test > SHADOW DOM 1`] = `
"<div class="hold">
  <div class="scroll-content">
    <div role="tablist">
      <slot name="tab">
      </slot>
    </div>
  </div>
  <span
    aria-hidden="true"
    class="separator"
  >
  </span>
  <span
    aria-hidden="true"
    class="indicator"
  >
  </span>
  <span
    aria-hidden="true"
    class="indicator"
  >
  </span>
</div>
<div>
  <slot name="tabpanel">
  </slot>
</div>
"
`;

exports[`BlockquoteTabs > Default > Semantic Dom and a11y > LIGHT DOM - Structure test > LIGHT DOM 1`] = `
"<blockquote-tabs>
  <blockquote-tab
    aria-selected="false"
    role="tab"
    slot="tab"
    tabindex="-1"
  >
    Tab 1
  </blockquote-tab>
  <blockquote-tab
    aria-selected="false"
    role="tab"
    slot="tab"
    tabindex="-1"
  >
    Tab 2
  </blockquote-tab>
  <blockquote-tab
    aria-selected="false"
    role="tab"
    slot="tab"
    tabindex="-1"
  >
    Tab 3
  </blockquote-tab>
  <blockquote-tabpanel
    aria-hidden="true"
    aria-labelledby="tab-1"
    role="tabpanel"
    slot="tabpanel"
    tabindex="0"
  >
    <p>
      Panel 1
    </p>
  </blockquote-tabpanel>
  <blockquote-tabpanel
    aria-hidden="true"
    aria-labelledby="tab-2"
    role="tabpanel"
    slot="tabpanel"
    tabindex="0"
  >
    <p>
      Panel 2
    </p>
  </blockquote-tabpanel>
  <blockquote-tabpanel
    aria-hidden="true"
    aria-labelledby="tab-3"
    role="tabpanel"
    slot="tabpanel"
    tabindex="0"
  >
    <p>
      Panel 3
    </p>
  </blockquote-tabpanel>
</blockquote-tabs>
"
`;

exports[`BlockquoteTabs > Default > Semantic Dom and a11y > SHADOW DOM - Structure test > SHADOW DOM 1`] = `
"<div class="hold">
  <div class="scroll-content">
    <div role="tablist">
      <slot name="tab">
      </slot>
    </div>
  </div>
  <span
    aria-hidden="true"
    class="separator"
  >
  </span>
  <span
    aria-hidden="true"
    class="indicator"
  >
  </span>
  <span
    aria-hidden="true"
    class="indicator"
  >
  </span>
</div>
<div>
  <slot name="tabpanel">
  </slot>
</div>
"
`;

exports[`BlockquoteTabs > Selected > Semantic Dom and a11y > LIGHT DOM - Structure test > LIGHT DOM 1`] = `
"<blockquote-tabs selected="2">
  <blockquote-tab
    aria-selected="false"
    role="tab"
    slot="tab"
    tabindex="-1"
  >
    Tab 1
  </blockquote-tab>
  <blockquote-tab
    aria-selected="true"
    role="tab"
    slot="tab"
    tabindex="0"
  >
    Tab 2
  </blockquote-tab>
  <blockquote-tab
    aria-selected="false"
    role="tab"
    slot="tab"
    tabindex="-1"
  >
    Tab 3
  </blockquote-tab>
  <blockquote-tabpanel
    aria-hidden="true"
    aria-labelledby="tab-1"
    role="tabpanel"
    slot="tabpanel"
    tabindex="0"
  >
    <p>
      Panel 1
    </p>
  </blockquote-tabpanel>
  <blockquote-tabpanel
    aria-hidden="false"
    aria-labelledby="tab-2"
    role="tabpanel"
    slot="tabpanel"
    tabindex="0"
  >
    <p>
      Panel 2
    </p>
  </blockquote-tabpanel>
  <blockquote-tabpanel
    aria-hidden="true"
    aria-labelledby="tab-3"
    role="tabpanel"
    slot="tabpanel"
    tabindex="0"
  >
    <p>
      Panel 3
    </p>
  </blockquote-tabpanel>
</blockquote-tabs>
"
`;

exports[`BlockquoteTabs > Selected > Semantic Dom and a11y > SHADOW DOM - Structure test > SHADOW DOM 1`] = `
"<div class="hold">
  <div class="scroll-content">
    <div role="tablist">
      <slot name="tab">
      </slot>
    </div>
  </div>
  <span
    aria-hidden="true"
    class="separator"
  >
  </span>
  <span
    aria-hidden="true"
    class="indicator"
  >
  </span>
  <span
    aria-hidden="true"
    class="indicator"
  >
  </span>
</div>
<div>
  <slot name="tabpanel">
  </slot>
</div>
"
`;
